pkg/types/validation/installconfig.go

```diff
@@ -996,5 +1008,91 @@ func validateFeatureSet(c *types.InstallConfig) field.ErrorList {
 		}()
 		allErrs = append(allErrs, field.NotSupported(field.NewPath("featureSet"), c.FeatureSet, sortedFeatureSets))
 	}
+
+	if c.FeatureSet == configv1.TechPreviewNoUpgrade {
+		if c.BareMetal != nil {
+			if c.BareMetal.LoadBalancer != nil {
+				if !validateLoadBalancer(c.BareMetal.LoadBalancer.Type) {
+					allErrs = append(allErrs, field.Invalid(field.NewPath("platform", "baremetal", "loadBalancer", "type"), c.VSphere.LoadBalancer.Type, "invalid load balancer type"))
+				}
+				if c.BareMetal.LoadBalancer.Type == configv1.LoadBalancerTypeOpenShiftManagedDefault && len(c.BareMetal.APIVIPs) == 0 {
+					allErrs = append(allErrs, field.Invalid(field.NewPath("platform", "baremetal", "apiVIPs"), c.BareMetal.APIVIPs, "must be set when using the OpenShiftManaged default load balancer"))
+				}
+			}
+		}
+		if c.Nutanix != nil {
+			if c.Nutanix.LoadBalancer != nil {
+				if !validateLoadBalancer(c.Nutanix.LoadBalancer.Type) {
+					allErrs = append(allErrs, field.Invalid(field.NewPath("platform", "nutanix", "loadBalancer", "type"), c.Nutanix.LoadBalancer.Type, "invalid load balancer type"))
+				}
+				if c.Nutanix.LoadBalancer.Type == configv1.LoadBalancerTypeOpenShiftManagedDefault && len(c.Nutanix.APIVIPs) == 0 {
+					allErrs = append(allErrs, field.Invalid(field.NewPath("platform", "nutanix", "apiVIPs"), c.Nutanix.APIVIPs, "must be set when using the OpenShiftManaged default load balancer"))
+				}
+			}
+		}
+		if c.OpenStack != nil {
+			if c.OpenStack.LoadBalancer != nil {
+				if !validateLoadBalancer(c.OpenStack.LoadBalancer.Type) {
+					allErrs = append(allErrs, field.Invalid(field.NewPath("platform", "openstack", "loadBalancer", "type"), c.OpenStack.LoadBalancer.Type, "invalid load balancer type"))
+				}
+				if c.OpenStack.LoadBalancer.Type == configv1.LoadBalancerTypeOpenShiftManagedDefault && len(c.OpenStack.APIVIPs) == 0 {
+					allErrs = append(allErrs, field.Invalid(field.NewPath("platform", "openstack", "apiVIPs"), c.OpenStack.APIVIPs, "must be set when using the OpenShiftManaged default load balancer"))
+				}
+			}
+		}
+		if c.Ovirt != nil {
+			if c.Ovirt.LoadBalancer != nil {
+				if !validateLoadBalancer(c.Ovirt.LoadBalancer.Type) {
+					allErrs = append(allErrs, field.Invalid(field.NewPath("platform", "ovirt", "loadBalancer", "type"), c.Ovirt.LoadBalancer.Type, "invalid load balancer type"))
+				}
+				if c.Ovirt.LoadBalancer.Type == configv1.LoadBalancerTypeOpenShiftManagedDefault && len(c.Ovirt.APIVIPs) == 0 {
+					allErrs = append(allErrs, field.Invalid(field.NewPath("platform", "ovirt", "apiVIPs"), c.Ovirt.APIVIPs, "must be set when using the OpenShiftManaged default load balancer"))
```

Comment:
What do you think of `field.Required` instead of `Invalid`? Invalid seems to be about a malformed value, while here it's empty and that's the problem.

Reply:
very good idea!